middleware/logic.go

```diff
@@ -0,0 +1,122 @@
+package middleware
+
+import (
+	"context"
+	"time"
+
+	"github.com/chihaya/chihaya/bittorrent"
+	"github.com/chihaya/chihaya/frontend"
+	"github.com/chihaya/chihaya/pkg/log"
+	"github.com/chihaya/chihaya/pkg/stop"
+	"github.com/chihaya/chihaya/storage"
+)
+
+// ResponseConfig holds the configuration used for the actual response.
+//
+// TODO(jzelinskie): Evaluate whether we would like to make this optional.
```

Comment:
How can we make this optional and still have a working tracker?

Reply:
That's the trade-off. We can make Chihaya extensible enough that you can program a new response generator at the cost of making config that won't compose a functional tracker.

Reply:
Oooh, that's what you mean! Yeah, makes sense.